### PR TITLE
Error code for Speedup Module

### PR DESCRIPTION
**Pull request by zheng-ningxin**
Add Error code for speedup module.

### CODE REVIEW

File: nni/compression/pytorch/speedup/error_code.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class SpeedupError(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __str__(self):
+        return str(self.msg)
+
+class EmptyLayerError(SpeedupError):
+    def __init__(self):
+        super(EmptyLayerError, self).__init__("Pruning a Layer to empty is not legal")
```
Comment (J-shang) [inline nni/compression/pytorch/speedup/error_code.py:13]: In this situation, should we suggest the user add this layer in `exclude`?

File: nni/compression/pytorch/speedup/error_code.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class ShapeMisMatchError(SpeedupError):
+    def __init__(self):
+        super(ShapeMisMatchError, self).__init__("Shape mismatch!")
```
Comment (J-shang) [inline nni/compression/pytorch/speedup/error_code.py:17]: maybe we can also add layer name in the error message for debugging?

Comment (zheng-ningxin) [reply nni/compression/pytorch/speedup/error_code.py:17]: Good suggestion, but I cannot get the `op_name` in the module replace functions under current interfaces. To keep the replacement function interface clean, I suggest to just report the error type now.

File: nni/compression/pytorch/speedup/error_code.py (hunk 1 above)
Comment (zheng-ningxin) [reply nni/compression/pytorch/speedup/error_code.py:13]: Not exactly the same,  there may be two reasons that a layer is pruned to empty: (1) it's sparsity ratio equals to 1.0 (2) it's output is useless after the mask propagation.

Comment (J-shang) [reply nni/compression/pytorch/speedup/error_code.py:13]: got it

**Review by J-shang — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]

**Pull request merged**